google-ads/build.gradle

```diff
@@ -275,3 +277,79 @@ task testShadowJar {
     }
   }
 }
+
+def configurePom(publication, nameToSet, descriptionToSet) {
+  publication.pom {
+    name = nameToSet
+    description = descriptionToSet
+    url = 'https://github.com/googleads/google-ads-java'
+    licenses {
+      license {
+        name = 'The Apache License, Version 2.0'
+        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+      }
+    }
+    scm {
+      connection = 'scm:git:https://github.com/googleads/google-ads-java'
+      developerConnection = 'scm:git:https://github.com/googleads/google-ads-java'
+      url = 'https://github.com/googleads/google-ads-java'
+    }
+    developers {
+      developer {
+        id = 'josh'
+        name = 'Josh Radcliff'
+        email = 'jradcliff@users.noreply.github.com'
+        organization = 'Google'
+        organizationUrl = 'https://www.google.com'
+      }
+      developer {
+        id = 'nick'
+        name = 'Nick Birnie'
+        email = 'jradcliff@users.noreply.github.com'
+        organization = 'Google'
+        organizationUrl = 'https://www.google.com'
+      }
+      developer {
+        id = 'devin'
+        name = 'Devin Chasanoff'
+        email = 'jradcliff@users.noreply.github.com'
+        organization = 'Google'
+        organizationUrl = 'https://www.google.com'
+      }
+    }
+  }
+}
+
+publishing {
+  publications {
+    shadow(MavenPublication) { publication ->
+      project.shadow.component(publication)
+
+      groupId = 'com.google.api-ads'
+      artifactId = 'google-ads-shadowjar'
+      version = '0.0.1-alpha'
+
+      configurePom(publication,
+          "Google Ads API client library for java - shadow jar",
+          "Provides a jar with all dependencies")
+    }
+  }
+
+  repositories {
+    maven {
+      url 'https://oss.sonatype.org/service/local/staging/deploy/maven2'
+
+      def credentialsConfig = file(System.properties['user.home'] + '/.sonatype-creds')
+      if (credentialsConfig.exists()) {
+        def props = new Properties()
+        credentialsConfig.withInputStream { props.load(it) }
+        credentials {
+          username props['username']
+          password props['password']
+        }
+      } else {
+        logger.warn("Unable to load Sonatype credentials from ~/.sonatype-creds. Publishing will fail.")
```

Comment:
I like having this warning, but I'm a little concerned about showing it to every user who runs any gradle task.

Can you instead move this to the first step of publishing? It could even be an error for that case.

Reply:
Added a check.

Reply:
Thanks. Confirmed I no longer get the warning for non-pub tasks.